### PR TITLE
Rr/batchnorm

### PR DESCRIPTION
**Pull request by rrawther**
WIP:  added scale_layer and modified batch_norm layer.
inferencegen: added WriteGDF changes for merging batchnorm and scale in caffe.proto using flag value 2.

### CODE REVIEW

File: utils/inference_generator/src/inference_generator.cpp [hunk 1]
```diff
@@ -1677,10 +1822,11 @@ int main(int argc, char* argv[])
     if(argc > 9) roundPolicy = argv[9];
     std::vector<std::vector<std::string>> net;
 
-    if(flags != 1 && flags != 0 ) {
-        printf("ERROR: flags can only take 0 or 1\n");
+    if(flags > 2 ) {
```
Comment (lcskrishna) [inline utils/inference_generator/src/inference_generator.cpp]: Check this error check again. if flags = -2 , bFuseScaleWithBatchNorm will be 1.

File: vx_nn/src/scale_layer.cpp [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    data->alpha = 1; data->beta = 0;
+
+    //input and output memory.
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_BUFFER_OPENCL, &data->input_mem, sizeof(data->input_mem)));
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[1], VX_TENSOR_BUFFER_OPENCL, &data->bnScale, sizeof(data->bnScale)));
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[2], VX_TENSOR_BUFFER_OPENCL, &data->bnBias, sizeof(data->bnBias)));
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[3], VX_TENSOR_BUFFER_OPENCL, &data->output_mem, sizeof(data->output_mem)));
+
+#if ENABLE_DEBUG_PRINT_DIMS
+    std::cout << "batch_normalization input " << input_dims[3] << " " << input_dims[2] << " " << input_dims[1] << " " << input_dims[0] << " ";
```
Comment (lcskrishna) [inline vx_nn/src/scale_layer.cpp]: try to change as scale_layer input

File: utils/inference_generator/src/inference_generator.cpp (hunk 1 above)
Comment (rgiduthuri) [reply utils/inference_generator/src/inference_generator.cpp]: It's better to remove the "if (flags > 2) ..." check completely.
The bit-0 of flags allows user to specify isSplitEnabled.
The bit-1 of flags allows user to specify bFuseScaleWithBatchNorm.
Other bits will be ignored.

Comment (rrawther) [reply utils/inference_generator/src/inference_generator.cpp]: Thanks for the suggestions. I will remove the check and get the LSB (2bits) using mask and ignore others.

File: vx_nn/src/scale_layer.cpp (hunk 2 above)
Comment (rrawther) [reply vx_nn/src/scale_layer.cpp]: Yes. I will.
